client/hosting/sites/components/sites-dashboard.tsx

```diff
@@ -98,10 +100,20 @@ const SitesDashboard = ( {
 	const [ initialSortApplied, setInitialSortApplied ] = useState( false );
 
 	const { hasSitesSortingPreferenceLoaded, sitesSorting, onSitesSortingChange } = useSitesSorting();
+	const sitesFilterCallback = ( site: SiteExcerptData ) => {
+		return ! site.options?.is_domain_only;
+	};
+
+	const getFilterArgs = ( siteType: string ): string[] => {
+		if ( siteType === 'all' ) {
+			return [ 'non-p2' ];
+		}
+		return [ 'p2' ];
+	};
 
 	const { data: allSites = [], isLoading } = useSiteExcerptsQuery(
-		[],
-		( site ) => ! site.options?.is_domain_only,
+		getFilterArgs( siteType ),
+		sitesFilterCallback,
```

Comment:
I think we might need to update the useSiteExcerptQuery. I noticed something weird, on my first load of /p2s the list initially showed both P2s and non-P2s for me (until this query fully updated it). I went looking at this and found this [initialData](https://github.com/Automattic/wp-calypso/blob/cc5e5eb7a18c9c94be1a2280a275277956b3ddc3/client/data/sites/use-site-excerpts-query.ts#L61) it returns. It returns all the sites that are in the store that are not nullish! P2 or not. We may need to update the hook to use the sitesFilterFn in intitial data (and update our filterFn for the p2s as well?) OR create another different but similar hook for this p2 case specifically? 🤔

Reply:
Ah I seen that but hadn't time yet to investigate it, thanks for the tip.